DisCatSharp/Clients/Rest/BaseMethods/ApplicationCommands/DiscordApiClient.BaseMethods.ApplicationCommands.cs

```diff
@@ -0,0 +1,472 @@
+			Options = command.Options,
+			NameLocalizations = command.NameLocalizations.GetKeyValuePairs(),
+			DescriptionLocalizations = command.DescriptionLocalizations.GetKeyValuePairs(),
+			DefaultMemberPermission = command.DefaultMemberPermissions,
+			DmPermission = command.DmPermission/*,
+			Nsfw = command.IsNsfw*/
+		};
+
+		var route = $"{Endpoints.APPLICATIONS}/:application_id{Endpoints.COMMANDS}";
+		var bucket = this.Rest.GetBucket(RestRequestMethod.POST, route, new {application_id = applicationId }, out var path);
+
+		var url = Utilities.GetApiUriFor(path, this.Discord.Configuration);
+		var res = await this.ExecuteRestRequest(this.Discord, bucket, url, RestRequestMethod.POST, route, payload: DiscordJson.SerializeObject(pld));
+
+		var ret = JsonConvert.DeserializeObject<DiscordApplicationCommand>(res.Response);
+		ret.Discord = this.Discord;
+
+		return ret;
+	}
+
+	/// <summary>
+	/// Gets a global application command.
+	/// </summary>
+	/// <param name="applicationId">The application id.</param>
+	/// <param name="commandId">The command id.</param>
+	internal async Task<DiscordApplicationCommand> GetGlobalApplicationCommandAsync(ulong applicationId, ulong commandId)
+	{
+		var route = $"{Endpoints.APPLICATIONS}/:application_id{Endpoints.COMMANDS}/:command_id";
+		var bucket = this.Rest.GetBucket(RestRequestMethod.GET, route, new {application_id = applicationId, command_id = commandId }, out var path);
+
+		var url = Utilities.GetApiUriFor(path, this.Discord.Configuration);
+		var res = await this.ExecuteRestRequest(this.Discord, bucket, url, RestRequestMethod.GET, route);
+
+		var ret = JsonConvert.DeserializeObject<DiscordApplicationCommand>(res.Response);
+		ret.Discord = this.Discord;
+
+		return ret;
+	}
+
+	/// <summary>
+	/// Edits a global application command.
+	/// </summary>
+	/// <param name="applicationId">The application id.</param>
+	/// <param name="commandId">The command id.</param>
+	/// <param name="name">The name.</param>
+	/// <param name="description">The description.</param>
+	/// <param name="options">The options.</param>
+	/// <param name="nameLocalization">The localizations of the name.</param>
+	/// <param name="descriptionLocalization">The localizations of the description.</param>
+	/// <param name="defaultMemberPermission">The default member permissions.</param>
+	/// <param name="dmPermission">The dm permission.</param>
+	/// <param name="isNsfw">Whether this command is marked as NSFW.</param>
+	internal async Task<DiscordApplicationCommand> EditGlobalApplicationCommandAsync(ulong applicationId, ulong commandId,
+		Optional<string> name, Optional<string> description, Optional<IReadOnlyCollection<DiscordApplicationCommandOption>> options,
+		Optional<DiscordApplicationCommandLocalization> nameLocalization, Optional<DiscordApplicationCommandLocalization> descriptionLocalization,
+		Optional<Permissions> defaultMemberPermission, Optional<bool> dmPermission, Optional<bool> isNsfw)
```

Comment:
Not just here but like.. everywhere 

What do we think about transforming long method signatures (aka lots of parameters) into something cleaner?

Reason being, we have so many parameters, sometimes they're not used (we pass null into them).
However, the order of these parameters is still "significant" despite not wanting certain things. Must admit that passing in multiple null values looks whack. 

For instance if we used records for immutability

```csharp
public record EditGlobalApplicationCommandRequest(
ulong ApplicationId,
ulong CommandId,
Optional<string> Name,
Optional<string> Description,
Optional<IReadOnlyCollection<DiscordApplicationCommandOption>> Options,
Optional<DiscordApplicationCommandLocalization> NameLocalization,
Optional<DiscordApplicationCommandLocalization> DescriptionLocalization,
Optional<Permissions> DefaultMemberPermissions,
Optional<bool> DmPermission,
Optional<bool> IsNSFW
);
```

Or, another approach would be to introduce a fluent-api methodology into the mix. Heck, a combination of the two... who knows